File: samples/sample.py
"""Sample file for testing copy-fn-signatures."""

import math
from dataclasses import dataclass


@dataclass
class Point:
    x: float
    y: float

    def distance(self,
                 other: "Point") -> float:
        return math.hypot(self.x - other.x, self.y - other.y)

    @staticmethod
    def origin() -> "Point":
        return Point(0.0, 0.0)

    @classmethod
    def from_polar(cls, r: float, theta: float) -> "Point":
        return cls(r * math.cos(theta), r * math.sin(theta))


def area_of_circle(radius: float) -> float:
    return math.pi * radius ** 2


def wrap(value, default=lambda: None):
    return default() if value is None else value

File: samples/sample.cs
using System;

namespace Demo
{
    public struct Point
    {
        public int X { get; set; }
        public int Y { get; set; }

        public void Move(int dx, int dy)
        {
            X += dx;
            Y += dy;
        }

        public override string ToString() => $"({X}, {Y})";
    }

    public interface IShape
    {
        double Area();
    }

    public class Circle : IShape
    {
        public double Radius { get; set; }

        public double Area() => Math.PI * Radius * Radius;
    }
}
